### ISSUE TITLE
Probably bug detected

### ISSUE THREAD

**Issue by P1119r1m** [OP-TEE/optee_os#1602]
I'm not almost sure, that here is a bug. Please correct me if I'm wrong.

Branch:
**master**

Commit:
**70bb1a1**

Place:
**optee_os/core/arch/arm/mm/core_mmu_v7.c**

Function:

```
void core_init_mmu_tables(struct tee_mmap_region *mm)
{
	void *ttb1 = (void *)core_mmu_get_main_ttb_va();
	size_t n;

	/* reset L1 table */
	memset(ttb1, 0, L1_TBL_SIZE);

	for (n = 0; mm[n].size; n++)
		if (!core_mmu_is_dynamic_vaspace(mm + n))
			map_memarea(mm + n, ttb1);
}
```

I think here it is not a bad idea to check if the element from memory table array is not of MEM_AREA_NOTYPE type.

```
void core_init_mmu_tables(struct tee_mmap_region *mm)
{
	void *ttb1 = (void *)core_mmu_get_main_ttb_va();
	size_t n;

	/* reset L1 table */
	memset(ttb1, 0, L1_TBL_SIZE);

	for (n = 0; mm[n].size; n++)
	{
		if(MEM_AREA_NOTYPE == (mm + n)->type)
			break;

		if (!core_mmu_is_dynamic_vaspace(mm + n))
			map_memarea(mm + n, ttb1);
	}
}
```

P.S. In my concrete situation, for an experimental ARMv7 platform I got an exception. And please try to forgive me if I'm wrong.

**Comment by jenswi-linaro** [comment]
The `mm` parameter is in practice a pointer to `static_memory_map`, which is supposed to be zero initialized so if `mm[n].size == 0` it should identify the last entry.

That said, I think it would be more correct for the for loop to be written as:
```
for (n = 0; mm[n].type != MEM_AREA_NOTYPE; n++)
...
```
 since that's how the last entry is identified in `init_mem_map()`.

A pull request for this is welcome.

**Comment by P1119r1m** [comment]
@jenswi-linaro Thanks for your response.  I don't understand what means "A pull request for this is welcome.". Could you please tell me what do you mean? I've tried to discover "Pull request" page from this site, but there is required 2 branches. Should I create one for this small fix? I'm not sure if I have the write permissions... Thanks!

**Comment by jenswi-linaro** [comment]
See https://github.com/OP-TEE/optee_os/blob/master/documentation/github.md

It's a good opportunity to learn how to contribute patches to OP-TEE. If it's too much trouble, I'll take care of it sooner or later.

**Comment by P1119r1m** [comment]
@jenswi-linaro I got it! Thank you!

**Comment by P1119r1m** [comment]
Closed by 432f64c commit.
